test/Generators/Microsoft.Gen.Logging/Unit/ParserTests.cs

```diff
@@ -966,6 +968,56 @@ static partial class C
             ");
     }
 
+    [Fact]
+    public void MultipleTypeDefinitions()
+    {
+        // Adding a dependency to an assembly that has internal definitions of public types
+        // should not result in a collision and break generation.
+        // Verify usage of the extension GetBestTypeByMetadataName(this Compilation) instead of Compilation.GetTypeByMetadataName().
+        var referencedSource = @"
+                namespace Microsoft.Extensions.Logging
+                {
+                    internal class LoggerMessageAttribute { }
+                }
+                namespace Microsoft.Extensions.Logging
+                {
+                    internal interface ILogger { }
+                    internal enum LogLevel { }
+                }";
+
+        // Compile the referenced assembly first.
+        Compilation referencedCompilation = CompilationHelper.CreateCompilation(referencedSource);
+
+        // Obtain the image of the referenced assembly.
+        byte[] referencedImage = CompilationHelper.CreateAssemblyImage(referencedCompilation);
+
+        // Generate the code
+        string source = @"
+                namespace Test
+                {
+                    using Microsoft.Extensions.Logging;
+
+                    partial class C
+                    {
+                        [LoggerMessage(EventId = 1, Level = LogLevel.Debug, Message = ""M1"")]
+                        static partial void M1(ILogger logger);
+                    }
+                }";
+
+        MetadataReference[] additionalReferences = { MetadataReference.CreateFromImage(referencedImage) };
+
+        Compilation compilation = CompilationHelper.CreateCompilation(source, additionalReferences);
+        LoggingGenerator generator = new LoggingGenerator();
+
+        (IReadOnlyList<Diagnostic> diagnostics, ImmutableArray<GeneratedSourceResult> generatedSources) =
+            RoslynTestUtils.RunGenerator(compilation, generator);
+
+        // Make sure compilation was successful.
+        Assert.Empty(diagnostics);
+        Assert.Single(generatedSources);
+        Assert.Equal(40, generatedSources[0].SourceText.Lines.Count);
```

Comment:
For blob assertions we should use VerifyXunit
https://github.com/dotnet/extensions/blob/0cd4e2e66a3b453c15943c3f9c242545775f706c/test/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring.Tests/Linux/LinuxUtilizationParserCgroupV2Tests.cs#L164

Reply:
Thanks for the tip. Had to include a scrubber for the assembly version.